apps/dotcom/client/src/tla/app/TldrawApp.ts

```diff
@@ -484,39 +484,33 @@ export class TldrawApp {
 		return this.getUserFileStates().find((f) => f.fileId === fileId)
 	}
 
-	updateFileState(fileId: string, cb: (fileState: TlaFileState) => Partial<TlaFileState>) {
+	updateFileState(fileId: string, partial: Partial<TlaFileState>) {
 		const fileState = this.getFileState(fileId)
 		if (!fileState) return
 		// remove relationship because zero complains
-		const { file: _, ...rest } = fileState
-		this.z.mutate.file_state.update({ ...cb(rest), fileId, userId: fileState.userId })
+		this.z.mutate.file_state.update({ ...partial, fileId, userId: fileState.userId })
 	}
 
 	async onFileEnter(fileId: string) {
 		await this.getOrCreateFileState(fileId)
-		this.updateFileState(fileId, (fileState) => ({
-			firstVisitAt: fileState.firstVisitAt ?? Date.now(),
```

Comment:
This was the only reason why `updateFileState` had to receive a callback. Since we can't mutate the `firstVisitAt` anyway I could remove it.